app/models/support/service_performance.rb

```diff
@@ -1,4 +1,318 @@
 class Support::ServicePerformance
+  def percentage_of_devolved_schools_that_have_signed_in
+    if number_of_devolved_schools.positive?
+      (number_of_devolved_schools_that_have_signed_in * 100.0 / number_of_devolved_schools).round
+    else
+      0
+    end
+  end
+
+  def number_of_devolved_schools_that_have_signed_in
+    @number_of_devolved_schools_that_have_signed_in ||=
+      School
+      .gias_status_open
+      .that_will_order_devices
+      .joins(user_schools: :user)
+      .where('users.sign_in_count > 0')
+      .count('DISTINCT(schools.id)')
+  end
+
+  def number_of_devolved_schools
+    @number_of_devolved_schools ||=
+      School
+      .gias_status_open
+      .that_will_order_devices
+      .count
+  end
+
+  def number_of_centrally_managed_schools
+    @number_of_centrally_managed_schools ||=
+      School
+      .gias_status_open
+      .that_are_centrally_managed
+      .count
+  end
+
+  def total_devices_available
+    SchoolDeviceAllocation.std_device.sum(:allocation)
+  end
+
+  def total_devices_ordered
+    SchoolDeviceAllocation.std_device.sum(:devices_ordered)
+  end
+
+  def total_devices_remaining
+    SchoolDeviceAllocation.std_device.sum('allocation - devices_ordered')
```

Comment:
```
SchoolDeviceAllocation.joins(:school).where(schools: { status: 'closed' }).sum('allocation - devices_ordered')
```
The above returns a non-zero number. do closed schools need to be excluded?

Reply:
I'm not sure, it depends on the meaning of the values we're displaying I guess. If we're showing the total devices then those closed schools that ordered still had devices that are presumably still in use and the successor school may well have had an adjustment to their allocation to account for it. Also we have to leave closed school in virtual pools at the moment so their figures are still used.